activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -167,6 +167,24 @@ def attribute_names
         end
       end
 
+      # Returns an Array of Strings that can be treated as column names. This
+      # includes attributes and attribute aliases.
+      #
+      #   class Person < ActiveRecord::Base
+      #   end
+      #
+      #   Person.effective_column_names
+      #   # => ["id", "created_at", "updated_at", "name", "age"]
+      def effective_column_names
+        @effective_column_names ||= begin
+          (attribute_names + attribute_aliases.keys).uniq
+        end
+      end
+
+      def effective_column_name?(*names)
```

Comment:
"Can I ask this model for attributes with these names?"

`Person.respond_to_attributes? :foo, :bar`